File: src/main/resources/js/matrix.js
function matrix(num) {

    // In ES6 we could create square 2D array as follows:
    // var results = [...Array(num).keys()].map(i => Array(num));

    // Create the square 2D array
    var results = [];

    for (var i = 0; i < num; i++) {
        results.push([]);
    }

    // The next value to insert
    var count = 1;
    var startRow = 0;
    var endRow = num - 1;
    var startCol = 0;
    var endCol = num - 1;

    while (startRow <= endRow && startCol <= endCol) {
        for (var i = startCol; i <= endCol; i++) {
            results[startRow][i] = count++;
        }

        startRow++;

        for (var i = startRow; i <= endRow; i++) {
            results[i][endCol] = count++;
        }

        endCol--;

        for (var i = endCol; i >= startCol; i--) {
            results[endRow][i] = count++;
        }

        endRow--;

        for (var i = endRow; i >= startRow; i--) {
            results[i][startCol] = count++;
        }

        startCol++;
    }

    return Java.to(results, "int[][]");
}

File: src/main/java/com/devxpress/Matrix.java
package com.devxpress;

public class Matrix {

    public static int[][] matrix(int num) {

        int[][] results = new int[num][num];

        int count = 1;
        int startRow = 0;
        int endRow = num - 1;
        int startCol = 0;
        int endCol = num - 1;

        while (startRow <= endRow && startCol <= endCol) {
            // Fill current start row
            for (int i = startCol; i <= endCol; i++) {
                results[startRow][i] = count++;
            }

            startRow++;

            // Fill current end column
            for (int i = startRow; i <= endRow; i++) {
                results[i][endCol] = count++;
            }

            endCol--;

            // Fill current end row
            for (int i = endCol; i >= startCol; i--) {
                results[endRow][i] = count++;
            }

            endRow--;

            // Fill current start column
            for (int i = endRow; i >= startRow; i--) {
                results[i][startCol] = count++;
            }

            startCol++;
        }

        return results;
    }

}
